src/fieldml_util_routines.f90

```diff
@@ -207,8 +207,6 @@ SUBROUTINE FIELDML_IO_FINALISE( FIELDML_INFO, ERR, ERROR, * )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_CONNECTIVITY_HANDLES, ERR, ERROR, *999 )
       CALL LIST_DESTROY( FIELDML_INFO%BASIS_LAYOUT_HANDLES, ERR, ERROR, *999 )
       DEALLOCATE(FIELDML_INFO)
```

Comment:
Yes, Caton wrote these routines and he didn't follow what had been done before. The data should be allocated in the initialise routine called from CreateStart etc. All allocations should be checked for error. We try not to dereference pointers across subroutine calls. Note that pointers can also have intent attributes. The intent means the same thing. INTENT(IN) means the pointer will not be changed. INTENT(OUT) means the pointer must be assigned and INTENT(INOUT) means both.
With regard to fixing the internal issues then I didn't mean for you to fix it. I was just pointing out that it was done incorrectly so that you didn't think that was how things are done. However, once you changed the finalise routine then the initialise also had to be changed to match. We can fix up all the other code later.